### PR TITLE
Update ids

### LINKED ISSUE #284: Google v Alphabet
Screenshot 2024-03-19 at 2 53 28 PM

How to handle Alphabet vs Google? Ex. is Alphabet (not Google) actually linked to 151 AI publications that are not tied to Google?

### LINKED ISSUE #295: Add ability to override raw company names
See also https://github.com/georgetown-cset/parat/issues/280 and https://github.com/georgetown-cset/parat/issues/244

### PR DESCRIPTION
**Pull request by jmelot**
Use old PARAT ids if available, otherwise use new ones plus an offset to prevent overlap (4000 is larger than any old PARAT id)

Closes #284 
Closes #295

### CODE REVIEW

File: company_linkage/parat_scripts/aggregate_organizations.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 # note: check https://docs.google.com/spreadsheets/d/1Tq28O8qIA6T3AJ5oTHKCcscaNZsY_E4OPOUm6JaiwWA/edit#gid=0
 # to ensure list is complete
 # we might switch this to a query or something instead of hard-coding it but this is easier for now
-no_roll_up = [101, 550, 1826, 313, 327, 2343]
+no_roll_up = [550, 1826, 313, 327, 2343]
```
Comment (rggelles) [inline company_linkage/parat_scripts/aggregate_organizations.py:11]: Is this a specific choice that we're now rolling up google into Alphabet?

File: company_linkage/sql/organizations.sql [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
+    new_cset_id
   FROM
     parat_input.groups
   where
     name = "Global 500"
+),
+mapped_parents AS (
+  SELECT
+    parentage.new_cset_id,
+    COALESCE(legacy_cset_id, 4000+parent_id) AS parent_id,
```
Comment (rggelles) [inline company_linkage/sql/organizations.sql:20]: Okay, so just trying to understand what's happening here:

We're linking the parent id (which has been changed here already by annotators to the new CSET ids? Is that correct?) to the organizations table, to find the org info about the parents. Then, we're finding the legacy cset id of the parent, and setting the parent id to be either the legacy id or 4000 + the new id. Meanwhile, we're leaving the cset id itself as the new_cset_id of the org itself, for now.

File: company_linkage/sql/organizations.sql [hunk 3]
```diff
@@ -47,7 +59,7 @@ SELECT
         UNNEST(market) m )) AS market )
 FROM (
   SELECT
-    CSET_id,
+    COALESCE(MAX(legacy_cset_id), 4000+new_cset_id) AS CSET_id,
```
Comment (rggelles) [inline company_linkage/sql/organizations.sql:62]: Then here we're taking the legacy cset id and coalescing it with 4000 + the new cset id to get the final cset id.

Okay, I think I follow!

File: web/scripts/retrieve_data.py [hunk 4]
```diff
@@ -52,7 +52,8 @@
     "江行智能": "Jiangxing Intelligence",
     "智易科技": "Zhiyi Tech",
     "创新奇智": "AInnovation",
-    "captricity": "Vidado"
+    "captricity": "Vidado",
+    "Alphabet": "Alphabet (including Google)"
```
Comment (rggelles) [inline web/scripts/retrieve_data.py:56]: Okay, yes, it looks like we've decided to roll up Google?

**Review by rggelles — APPROVED** [review]
LGTM! Put some comments in mostly just trying to understand the changes made, but everything looks correct.

File: company_linkage/sql/organizations.sql (hunk 2 above)
Comment (jmelot) [reply company_linkage/sql/organizations.sql:20]: Yep - the parent id is the "new cset id" for the parent organization - we're changing it here to match the ids that will be used in `organizations`. We're keeping the new cset id around for joining purposes

File: web/scripts/retrieve_data.py (hunk 4 above)
Comment (jmelot) [reply web/scripts/retrieve_data.py:56]: Yep, see https://github.com/georgetown-cset/parat/issues/284

**Pull request merged**